tests/data/exagridos.json

```diff
@@ -0,0 +1,2037 @@
+{
+    "os": {
+        "discovery": {
+            "devices": [
+                {
+                    "sysName": "<private>",
+                    "sysObjectID": ".1.3.6.1.4.1.14941.3.1.840",
+                    "sysDescr": "ExaGrid EX40000E-SEC, 4.8.1.1209.P60",
+                    "sysContact": null,
+                    "version": null,
+                    "hardware": null,
+                    "features": null,
+                    "os": "exagridos",
+                    "type": "storage",
+                    "serial": null,
+                    "icon": "exagrid.png",
+                    "location": null
+                }
+            ]
+        },
+        "poller": {
+            "devices": [
+                {
+                    "sysName": "<private>",
+                    "sysObjectID": ".1.3.6.1.4.1.14941.3.1.840",
+                    "sysDescr": "ExaGrid EX40000E-SEC, 4.8.1.1209.P60",
+                    "sysContact": "<private>",
+                    "version": null,
+                    "hardware": null,
```

Comment:
Seems that version is not collected, nor HW. You can probably check in the snmpwalk data from the device, and get them from an OID, if you have no good way to extract them from SysDescr.

Reply:
snmpwalk doesn't show there version or hw in another field than the sysDescr field.  If there is a way to separate them out in exagrid.inc.php, it is beyond me.

RFC1213-MIB::sysDescr.0 = STRING: "ExaGrid EX40000E-SEC, 4.8.1.1209.P60"

Reply:
OK. Then you probably need to adapt ``` includes/polling/os/exagrid.inc.php ``` to collect the version from here.

Reply:
If you can tell me how to do that, great.  If not, just close this.  Just trying to help contribute.  It shows the version and hardware version in my setup.

Reply:
you have code in ``` includes/polling/os/exagrid.inc.php  ```to fill the $version, which tries to parse sysDescr to extract it. But the Regexp does not match. 

So either you remove ``` includes/polling/os/exagrid.inc.php ``` (it does not do anything right now in your environnement) or patch it to actually match the bold part of sysDescr : 

ExaGrid EX40000E-SEC, **4.8.1.1209.P60**

Something like the code below should extract both Hardware and version to the proper variables. 
```
preg_match('/^ExaGrid ([^,]*), (.*)$/', $device['sysDescr'], $data);
$version = $data[2];
$hardware = $data[1];
```

Reply:
Thanks,  This is not my specialty.  Took me some time to get adding the device figured out.  I tried removing the php file all together and it didn't match.  Updated to what you had and renamed the file to exagridos.php and it is matching  now.  Will try to upload shortly.  

#### Load poller module os ####
Hardware: EX40000E-SEC
Version: 4.8.1.1209.P60
Features: 
Serial: